doc-content/enterprise-only/project-data/variables-con.adoc

```diff
@@ -0,0 +1,94 @@
+
+If the variable cannot be found, a read access request returns `null` and a write access produces an error message, and the process continues its execution.
+Variables are searched for based on their ID.
+
+== Defining global variables
+Global variables exist in a knowledge session and can be accessed and are shared by all assets in that session. They belong to the particular session of the Knowledge Base and they are used to pass information to the engine. Every global variable defines its ID and item subject reference. The ID serves as the variable name and must be unique within the process definition. The item subject reference defines the data type the variable stores.
+
+[IMPORTANT]
+====
+The rules are evaluated at the moment the fact is inserted.
+Therefore, if you are using a global variable to constrain a fact pattern and the global is not set, the system returns a ``NullPointerException``.
+====
+
+Global variables are initialized either when the process with the variable definition is added to the session or when the session is initialized with globals as its parameters.
+
+Values of global variables can typically be changed during the assignment, which is a mapping between a process variable and an activity variable. The global variable is then associated with the local activity context, local activity variable, or by a direct call to the variable from a child context.
+
+.Prerequisites
+* You have created a project in {CENTRAL} and it contains at least one business process asset.
+
+.Procedure
+. Open a business process asset.
+. Click a blank area of the process designer canvas.
+. Click the *Properties* icon on the upper-right side of the screen to open the *Properties* panel.
+. If necessary, expand the *Process* section.
+. In the *Global Variables* sub-section, click the plus icon.
+. Enter a name for the variable in the *Name* box.
+. Select a data type from the *Data Type* menu.
+
+== Defining process variables
+Process variables are defined as properties in the BPMN2 definition file and are visible within the process instance. They are initialized at process creation and destroyed on process completion.
+
+A process variable is a variable that exists in a process context and can be accessed by its process or its child elements. Process variables belong to a particular process instance and cannot be accessed by other process instances.
+Every process variable defines its ID and item subject reference: the ID serves as the variable name and must be unique within the process definition.
+The item subject reference defines the data type the variable stores.
+
+Process variables are initialized when the process instance is created.
+Their value can be changed by the process activities using the Assignment, when the global variable is associated with the local Activity context, local Activity variable, or by a direct call to the variable from a child context.
```

Comment:
Minor, but shouldn't the second sentence be right after the 1st?

Reply:
Yes, fixed.